### ISSUE TITLE
The S740 scanner loses connection when another application that uses the SDK is opened (iPadOS 16.5)

### ISSUE THREAD

**Issue by KevinRondon** [SocketMobile/cocoapods-capturesdk#6]
Since the iPadOS 16.5 update, I have been experiencing issues connecting the S740 scanner to different applications that use the Socket Mobile SDK. I conducted an experiment with the following steps while having the Socket in Application Mode:

- Download the SM Keyboard application and Socket Mobile Companion.
- Open the SM Keyboard application and wait for the reader to connect. It connects without any problem.
- Open the Socket Mobile Companion application. It doesn't connect.
- Reopen the SM Keyboard application, and the scanner disconnects and doesn't reconnect. It is necessary to close the other app and power cycle the scanner to reconnect it.

This issue occurs with any two apps that use the SDK.

I would greatly appreciate your prompt assistance as we rely on your integration for our company's operations and use the Square app, which immediately replicates the issue, causing disruptions in our clients' POS systems. I am available to provide any additional information necessary to replicate the error.

It is important to mention that we don't experience this issue on previous versions of iOS.

**Comment by cyrille-socket** [comment]
Hi @KevinRondon ,

it seems like iPadOS 16 introduced some Bluetooth issues and especially since 16.5.

Can you confirm that your issue is on iPadOS and not on iOS please?

Cyrille

**Comment by KevinRondon** [comment]
Hi @cyrille-socket ,

You are right, it's on iPadOS 16.5. No further information about the iPadOS Bluetooth problem?

Thank you for your response and attention

**Comment by cyrille-socket** [comment]
Hi @KevinRondon 

we are still investigating on the issue but it seems that something has changes in iPadOS 16.5 causing this. It's still a bit unclear to us at the moment.

Keep you posted.

Cyrille

**Comment by cyrille-socket** [comment]
Hi @KevinRondon 

so good news, we have identified the issue and will provide a new SDK version on Monday next week.

Seems like  changed something in the ExternalAccessory framework to handle Bluetooth connections...

**Comment by KevinRondon** [comment]
Hi @cyrille-socket 

Those are fantastic news! I really appreciate you keeping me updated with the information. I will be fully attentive to the Monday update. You have no idea how happy it makes us to hear this.

I understand, it makes sense. Other external devices to Apple have also caused us some issues with Bluetooth connectivity in this new version.

Thank you very much for your work and attention. Integration with you is vital for us.

**Comment by cyrille-socket** [comment]
Hi @KevinRondon 

we just released the new version of CaptureSDK with the fix for Bluetooth connection due to iPadOS 16.5.

https://github.com/SocketMobile/cocoapods-capturesdk/releases/tag/1.6.39

Let me know if that works for you.

Cyrille

**Comment by KevinRondon** [comment]
Hi @cyrille-socket 

It seems to be working fine! We updated two of our apps with the new version, and they connect without any issues when switching between them. However, we are experiencing conflicts with a third-party app (Square POS) that also uses the SDK. When we open their app, the problem arises. Nevertheless, we have already informed them about the need to update their SDK. 

Thank you very much for your update and attention.

**Comment by KevinRondon** [comment]
iPadOS 16.5+ disconnection problem still occurs when switching apps fast.

Hello @cyrille-socket  again!
We managed to get Square to update their application with the new SDK version where you resolved the disconnection issue between different apps when using the SDK. In simple tests, it worked with our app. However, our customers started experiencing the problem again intermittently. Therefore, we started investigating, and one of the things we found is that if you switch between applications before it connects in one of them, the problem resurfaces, and the socket remains completely disconnected (when the socket is disconnected, it also affects other Bluetooth devices that we have paired in the app, such as Star Micronics printers, this doesn't happen if we don't have the socket connected at any time, which is quite serious for us).

In short, quick app switching replicates the problem. Tomorrow, I'll have the devices at my disposal, in case you need more information on how to replicate the issue.

**Comment by cyrille-socket** [comment]
Hello @KevinRondon 

that's quite unfortunate. At the moment I don't reproduce this behaviour. 

The only thing I can tell you is that Star does not seem to have updated its SDK since quite some time so they also should address this issue by closing the connection of their Bluetooth classic printers.

We'll continue to investigate and keep you posted

**Comment by KevinRondon** [comment]
Hey @cyrille-socket,

The problem is intermittent, sometimes it's easier to replicate than others. I conducted some tests recorded on video to show the issue. These tests were done with only the Socket Mobile connected (nothing from Star was connected, and the devices were forgotten, meaning the only device paired with the iPad is the Socket).

The iPad version is 16.5.1.

[For the first test](https://drive.google.com/file/d/1KkahqL1oyIyP6BgAwcXvZgy0fJsdWiJz/view?usp=sharing), I used the following Socket Mobile applications:

SM Keyboard by Socket Mobile 2.0.5
Socket Mobile Companion 3.1.3

As you can see, when interacting rapidly between the applications, there's a moment when the device gets disconnected and doesn't connect in either of the two applications. Most of the time, turning off the Socket and turning it back on again makes it work again, but there have been cases where I had to restart the iPad.

[For the second test](https://drive.google.com/file/d/1KlLTIi66KsmHUa-IvurC321p-VuTQdKG/view?usp=sharing), I used the apps that our clients use, and the same issue occurs. Among switching apps, there's a moment when the connection is lost in the apps.

[Videos with the tests](https://drive.google.com/drive/folders/1Kh5z6qfvtzUHKR7Vzs77dxwFx7Wi67MW?usp=sharing)

**Comment by KevinRondon** [comment]
Hi @cyrille-socket ,
I wanted to ask if there is any update. Were you able to replicate the issue? For us, integration with Socket is currently a major problem with our clients, and we have had to find solutions by integrating with other companies. However, we really want to continue supporting Socket Mobile on our platform. Any update on the issue would be appreciated by us.

**Comment by socketPM** [comment]
@KevinRondon Sorry we don't have further updates yet. But I want to add a few points:

- Switching between two Capture apps with scanner connected in neither app seems an unlikely real world use case. And in this scenario, you will need to at least switch to another app and then back quickly enough in order to lose the scanner ownership completely. 
- Have you got any insight from your customers reporting the issue that also ALL have a Star printer (or other Bluetooth Classic devices) connected? So far from what we learned this is an issue with Apple's External Accessory Framework. So any SDK on the framework, if not implementing a fix like ours, will likely cause the issue, not only for the particular device that is using that SDK, but for all the devices that connected to iPad through the External Accessory Framework. 
- On a side note, the SM Keyboard is a somewhat different animal. When it is placed in the mix, even before we saw the iPad issue, it could cause some ownership conflict with other Capture apps, because the way an app extension works. But loosk like you only used the main app, not the actual keyboard, so it should be less of a concern. 

Thanks,

**Comment by KevinRondon** [comment]
Thank you for your response @socketPM 

- In our workflow, we use the Square application to process payments through the Square POS API. This means that in our application, we redirect the cashier to the Square application, the payment is made, and then it automatically returns to our application. In the case where the payment is done quickly enough or the payment attempt is canceled, the problem occurs if the Socket Scanner has not yet been connected to the Square POS. (The scanner connection sometimes takes longer than others, that's why I mentioned the intermittence.)

- There is a client who doesn't have any Star printer and still experiences the connection issue. The rest of the clients who are having problems have stopped using the Socket Scanner and haven't had any issues with other external devices such as Star printers. However, when they use the Socket Scanner again, the problem reoccurs. The tests I sent you were done without any other device connected, so it's entirely possible for the issue to occur solely with the Socket Scanner.

**Comment by cyrille-socket** [comment]
Hi @KevinRondon 

from our internal tests and several other developers feedback, iPadOS 16.6 solved the issue.

In our SDK, the additional work done to close the Bluetooth Classic connection after putting an app in background is still effective and does not harm.

Let me know if you already updated your device and advised your customers to do so.

Cyrille

**Comment by cyrille-socket** [comment]
Hi @KevinRondon 

I'm gonna close this issue. In case you still have issues with BT connection, please reopen it.

But last Apple update fixes the issue following all the feedback from developers we had now.

Best,
Cyrille
